### PR TITLE
Add workspace empty UI and connect to the server

### PR DESCRIPTION
**Pull request by montakarlo**

### CODE REVIEW

File: frontend/src/components/AddWorkspace/index.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import React, { useState, FunctionComponent, JSXElementConstructor, FunctionComponentElement } from 'react';
```
Comment (vlvereta) [inline frontend/src/components/AddWorkspace/index.tsx:1]: It's not a component, replace to the containers

File: frontend/src/reducers/index.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import UserReducer from './user';
 
 const rootReducer = () => combineReducers({
   router: connectRouter(history),
-  UserReducer
+  UserReducer,
```
Comment (vlvereta) [inline frontend/src/reducers/index.ts:8]: No Reducer part in the reducer names.

File: frontend/src/scenes/Workspace/reducer.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { Routine } from 'redux-saga-routines';
+import { postWorkspaceNameRoutine } from '../Workspace/routines'
+interface IWorkspaceState{
```
Comment (vlvereta) [inline frontend/src/scenes/Workspace/reducer.ts:3]: Add an empty line.

File: frontend/src/services/workspaceService.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import api from '../common/helpers/apiHelper'
+
+export async function addWorkspace(name: string) {
+  // const response = await makeRequest('https://',{
+  //     method: 'POST',
+  //     headers: {'Content-Type': 'application/json;charset=utf-8'},
+  //     body: JSON.stringify(name)
+  //   });
```
Comment (vlvereta) [inline frontend/src/services/workspaceService.ts:8]: Do you need this comment?

File: frontend/src/services/workspaceService.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  const response = await api.post('https://', {name})
+  return await response
```
Comment (vlvereta) [inline frontend/src/services/workspaceService.ts:10]: No need to use await in return statement.

File: frontend/src/scenes/Workspace/sagas.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { postWorkspaceNameRoutine } from './routines'
+import { Routine } from 'redux-saga-routines';
+import {takeEvery, put, call} from 'redux-saga/effects'
+import {addWorkspace} from '../../services/workspaceService'
+
+function* addWorkspaceReq({ payload }: Routine<any>) {
```
Comment (vlvereta) [inline frontend/src/scenes/Workspace/sagas.ts:6]: Fix any.

File: frontend/src/scenes/Workspace/routines/index.ts [hunk 7]
```diff
@@ -0,0 +1,3 @@
+import { createRoutine } from 'redux-saga-routines';
+
+export const postWorkspaceNameRoutine = createRoutine('ADD_WORKSPACE');
```
Comment (vlvereta) [inline frontend/src/scenes/Workspace/routines/index.ts:3]: Better to use name the same way as you action -> addWorkspaceRoutine, for example.

File: frontend/src/containers/AddWorkspace/index.tsx [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+import styles from './styles.module.sass';
+import logo from '../../img/chatitoTemp.png';
+import bgImg from '../../img/bg-img.png';
+import { connect } from 'react-redux';
+import { addWorkspaceRoutine } from '../../scenes/Workspace/routines';
+
+type IFetchWorkspace<T, S> = (name: T) => S;
+
+interface IProps {
+  addWorkspace: IFetchWorkspace<string, void>;
```
Comment (vlvereta) [inline frontend/src/containers/AddWorkspace/index.tsx:13]: Let's use it in the component

File: frontend/src/containers/AddWorkspace/styles.module.sass [hunk 9]
```diff
@@ -0,0 +1,81 @@
+.AddWorkspace
+  font-family: 'Recursive'
```
Comment (vlvereta) [inline frontend/src/containers/AddWorkspace/styles.module.sass:2]: Why do you use this font?

File: frontend/src/containers/AddWorkspace/styles.module.sass [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+  width: 100%
+  height: 100vh
+  display: flex
+  justify-content: center
+  align-items: center
+  &__container
+    width: 1024px
+    height: 75vh
+    // background-color: red
```
Comment (vlvereta) [inline frontend/src/containers/AddWorkspace/styles.module.sass:11]: Remove?

File: frontend/src/containers/AddWorkspace/styles.module.sass [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+    letter-spacing: 0px
+    line-height: 1.2
+    margin-bottom: 12px
+    font-weight: 600
+
+.inputButton
+  width: 130px
+  font-size: 22px
+  margin-bottom: 20px
+  background-color: #d1d0d0
```
Comment (vlvereta) [inline frontend/src/containers/AddWorkspace/styles.module.sass:48]: Colors should be imported from colors.ts

File: frontend/src/reducers/index.ts [hunk 12]
```diff
@@ -1,11 +1,14 @@
...
 import { history } from '../common/helpers/historyHelper';
+import UserReducer from './user';
+import WorkspaceReducer from '../scenes/Workspace/reducer';
 import user from './user';
 
 const rootReducer = () => combineReducers({
   router: connectRouter(history),
-  user
+  user,
+  WorkspaceReducer
```
Comment (vlvereta) [inline frontend/src/reducers/index.ts:11]: Don't use such names for store

File: frontend/src/services/workspaceService.ts [hunk 13]
```diff
@@ -0,0 +1,6 @@
+import api from '../common/helpers/apiHelper'
+
+export async function addWorkspace(name: string) {
+  const response = await api.post('https://', {name})
+  return response
+}
```
Comment (vlvereta) [inline frontend/src/services/workspaceService.ts:6]: Add an empty line

File: frontend/src/containers/AddWorkspace/styles.module.sass (hunk 11 above)
Comment (montakarlo) [reply frontend/src/containers/AddWorkspace/styles.module.sass:46]: Don't see this file

File: frontend/src/services/workspaceService.ts [hunk 14]
```diff
@@ -0,0 +1,11 @@
+import api from '../common/helpers/apiHelper'
+
+export async function addWorkspace(name: string) {
+  const response = await api.post('https://', {name})
```
Comment (AlikRakhmonov) [inline frontend/src/services/workspaceService.ts:4]: https?

Comment (AlikRakhmonov) [reply frontend/src/services/workspaceService.ts:4]: space? { name }

File: frontend/src/containers/Routing/index.tsx [hunk 15]
```diff
@@ -36,6 +37,7 @@ const Routing: React.FC<IProps> = ({
   return (
     <LoaderWrapper loading={isLoading || (hasToken && !isAuthorized)}>
       <Header />
+      <AddWorkspace />
```
Comment (vlvereta) [inline frontend/src/containers/Routing/index.tsx:40]: Add workspace scene should be available on `/w/add` route.

File: frontend/src/styles/colors.sass [hunk 16]
```diff
@@ -2,3 +2,4 @@
   --user-detector-online: yellowgreen
   --user-detector-offline: grey
   --user-detector-border: black
+  --background-color: #F7B61A
```
Comment (vlvereta) [inline frontend/src/styles/colors.sass:5]: Name it such as main-background-color if it's the main or specify it.

File: frontend/src/scenes/Workspace/styles.module.sass [hunk 17]
```diff
@@ -0,0 +1,6 @@
+.workspace
+  width: 100%
+  height: 100vh
+  display: flex
+  justify-content: center
+  align-items: center
```
Comment (vlvereta) [inline frontend/src/scenes/Workspace/styles.module.sass:6]: Empty line.

File: frontend/src/scenes/Workspace/styles.module.sass [hunk 18]
```diff
@@ -0,0 +1,6 @@
+.workspace
+  width: 100%
+  height: 100vh
```
Comment (vlvereta) [inline frontend/src/scenes/Workspace/styles.module.sass:3]: It seems that we have full page global style, am I right?

File: frontend/src/scenes/Workspace/reducers/reducer.ts [hunk 19]
```diff
@@ -0,0 +1,33 @@
+import { Routine } from 'redux-saga-routines';
+import { addWorkspaceRoutine } from '../routines';
+
+interface IWorkspaceState{
+  name: string;
```
Comment (vlvereta) [inline frontend/src/scenes/Workspace/reducers/reducer.ts:5]: Default structure: { data: IWorkspace, loading: boolean, error: ..}

File: frontend/src/scenes/Workspace/reducers/reducer.ts [hunk 20]
```diff
@@ -0,0 +1,33 @@
+import { Routine } from 'redux-saga-routines';
+import { addWorkspaceRoutine } from '../routines';
+
+interface IWorkspaceState{
```
Comment (vlvereta) [inline frontend/src/scenes/Workspace/reducers/reducer.ts:4]: Replace it closer to the Store.

File: frontend/src/scenes/Workspace/containers/WorkspaceToolbar/sagas.ts [hunk 21]
```diff
@@ -0,0 +1,23 @@
...
+
+function* watchFetchWorkspaces() {
+  yield takeEvery(fetchWorkspacesRoutine.TRIGGER, fetchWorkspaces)
+}
+
+export default function* WorkspaceToolbarSaga() {
+  yield all ([
+    watchFetchWorkspaces()
+  ])
+}
```
Comment (vlvereta) [inline frontend/src/scenes/Workspace/containers/WorkspaceToolbar/sagas.ts:23]: Empty line

**Pull request merged**